src/Control/Lens/Unsound.hs

```diff
@@ -5,10 +5,6 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# OPTIONS_GHC -fno-warn-warnings-deprecations #-}
 
-#ifndef MIN_VERSION_mtl
```

Comment:
I wonder if all these `#ifndef` statements should go into `include/lens-common.h` at some point... but let's not do so in this PR, since there's already enough code churn in this commit as it is.